data/policies/derive.cas

```diff
@@ -34,3 +43,9 @@ domain some_domain {
 	strategy_foo.read(this);
 	custom_define.read(this);
 }
+
+@derive([some_associated_call], strategy=union)
+virtual resource to_associate inherits foo, bar {}
+
+@associate([to_associate])
```

Comment:
I guess there should be a resulting `call associates-to_associate-some_associated_call` because of the `@associate` annotation, right?

Reply:
Yes....

It looks like the current CIL has `(call to_associate-some_associated_call (to_associate associates))`, which I think is wrong.  The function name `associates-to_associate-some_associated_call` is neither declared or called, and I think it should be.  Let me look into that some more and see what's going on.

Reply:
Alright, there were a couple of interacting bugs down there, so it was a pain to track down, but I believe I've gotten it all sorted out, and the call you mentioned now appears, with the erroneous one I mentioned gone.

The two bugs were:

* The comparison of which function name to call used the name of the resource being associated rather than the synthetic resource name
* Synthetic resources created by association cleared all annotations, which meant they didn't actually end up deriving after all.  I changed that to have them keep @derive and @makelist, which I think makes sense.  @alias isn't really possible since it defines a unique name that can't refer to two things.  @associate probably wouldn't hurt anything since it's already been processed at that point, but for the same reason, there's no point in keeping it.  For future annotation additions, we'll need to keep that list updated, so it should probably be generalized somehow, but I'll leave that aspect for future enhancements.

With this latest change, I believe I've addressed all your comments, either by fixing something or responding to the comment.  Would appreciate another round of feedback and your responses to my responses when you have time.